packages/maskbook/src/plugins/Trader/types/dodo.ts

```diff
@@ -0,0 +1,38 @@
+import type { FungibleTokenDetailed } from '@masknet/web3-shared'
+
+export interface SwapRouteRequest {
+    isNativeSellToken: boolean
+    fromToken: FungibleTokenDetailed
+    toToken: FungibleTokenDetailed
+    fromAmount: string
+    slippage: number
+    userAddr: string
+    rpc?: string
+    chainId: number
+}
+
+export interface SwapRouteResponse {
+    status: number
+    data: SwapRouteData
+}
+
+export interface SwapRouteData {
+    data: string
+    priceImpact: number
+    resAmount: number
+    resCostGas: number
+    resPricePerFromToken: number
+    resPricePerToToken: number
+    targetApproveAddr: string
+    targetDecimals: number
+    to: string
+    useSource: string
+    fromAmount: number
+    value: string
+}
+
+export interface SwapRouteErrorResponse {
+    code?: number
+    status?: number
+    data: string
+}
```

Comment:
```suggestion
export type SwapRouteResponse = SwapRouteSuccessResponse | SwapRouteErrorResponse

export interface SwapRouteSuccessResponse {
    code: 200
    status: number
    data: SwapRouteData
}

export interface SwapRouteErrorResponse {
    code: number
    status: number
    data: string
}

export interface SwapRouteData {
    data: string
    priceImpact: number
    resAmount: number
    resCostGas: number
    resPricePerFromToken: number
    resPricePerToToken: number
    targetApproveAddr: string
    targetDecimals: number
    to: string
    useSource: string
    fromAmount: number
    value: string
}
```

Reply:
```ts
const payload: SwapRouteResponse = await response.json()
if (payload.code !== 200) {
    throw new Error(payload.data ?? 'Unknown Error')
}
/* ... success logic */
```

Reply:
see https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes-func.html#discriminated-unions